sklearn/neighbors/_graph.py

```diff
@@ -281,7 +284,8 @@ class KNeighborsTransformer(KNeighborsMixin, UnsupervisedMixin,
     ...     KNeighborsTransformer(n_neighbors=5, mode='distance'),
     ...     Isomap(neighbors_algorithm='precomputed'))
     """
-    def __init__(self, mode='distance', n_neighbors=5, algorithm='auto',
+    @_deprecate_positional_args
+    def __init__(self, mode='distance', n_neighbors=5, *, algorithm='auto',
```

Comment:
I'd leave no positional here. `KNeighborsTransformer('distance', 5)` doesn't really read well.

Reply:
Yeah, we probably should have put n_neighbors first.